app/controllers/new-signup.js

```diff
@@ -80,25 +134,64 @@ export default class NewSignUpController extends Controller {
       return (this.error = ERROR_MESSAGES.userName);
     }
 
-    registerUser(signupDetails)
-      .then((res) => {
-        if (res.status === 204) {
-          this.analytics.identifyUser();
-          this.analytics.trackEvent(NEW_SIGNUP_FLOW.USER_REGISTERED);
-          this.currentStep = this.LAST_STEP;
-        } else {
-          this.analytics.trackEvent(NEW_SIGNUP_FLOW.UNABLE_TO_SIGNUP);
+    if (this.dev) {
+      fetch(`${BASE_API_URL}/users/self`, {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        credentials: 'include',
+      })
```

Comment:
Why fetch users/self, to get the roles and overwrite it with the changed roles?

If yes then what do you think about handling this from the backend itself?

Reply:
Want to discuss with you about this
I also thinking about this